Annex60/Utilities/Time/CalendarTime.mo

```diff
@@ -0,0 +1,279 @@
+within Annex60.Utilities.Time;
+model CalendarTime
+  "Computes the unix time stamp and calendar time from the simulation time"
+  extends Modelica.Blocks.Icons.Block;
+  parameter Annex60.Utilities.Time.BaseClasses.TimeReference timRef;
+  parameter Integer yearRef(min=firstYear, max=lastYear) = 2016
+    "Year when time = 0"
+    annotation(Dialog(enable=timRef==Annex60.Utilities.Time.BaseClasses.TimeReference.Custom));
+  // fixme: The xxxRef is dangerous and should be removed.
+  // The problem is that users may use them, and then the simulation time
+  // time=0 is somewhere different than Jan. 1, 0:00. In this case,
+  // ReaderTMY3 and the computations of the sun position, and maybe
+  // other data readers that use the convention that t=0 is Jan 1, midnight,
+  // will all produce wrong results. All building simulation programs I
+  // know of do use t=0 as midnight Jan 1, and then compute the actual time
+  // based on startTime.
+  parameter Integer monthRef(min=1, max=12) = 1 "Month when time = 0"
+    annotation(Dialog(enable=timRef==Annex60.Utilities.Time.BaseClasses.TimeReference.Custom));
+  parameter Integer dayRef(min=1, max=31) = 1 "Day when time = 0"
+    annotation(Dialog(enable=timRef==Annex60.Utilities.Time.BaseClasses.TimeReference.Custom));
+  parameter Integer hourRef(min=0, max=23) = 0 "Hour when time = 0"
+    annotation(Dialog(enable=timRef==Annex60.Utilities.Time.BaseClasses.TimeReference.Custom));
+  parameter Integer minuteRef(min=0, max=59) = 0 "Minute when time = 0"
+    annotation(Dialog(enable=timRef==Annex60.Utilities.Time.BaseClasses.TimeReference.Custom));
+  parameter Integer secondRef(min=0, max=59) = 0 "Second when time = 0"
+    annotation(Dialog(enable=timRef==Annex60.Utilities.Time.BaseClasses.TimeReference.Custom));
+    // fixme: units are missing
+  Modelica.Blocks.Interfaces.RealInput tim "Simulation time"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+    // fixme: units are missing
+  Modelica.Blocks.Interfaces.RealOutput unixTimeStamp
+    "Unix time stamp at GMT+0"
+        annotation (Placement(transformation(extent={{100,-70},{120,-50}})));
+  discrete Modelica.Blocks.Interfaces.IntegerOutput year "Year"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+  discrete Modelica.Blocks.Interfaces.IntegerOutput month "Month of the year"
+    annotation (Placement(transformation(extent={{100,10},{120,30}})));
+  Modelica.Blocks.Interfaces.IntegerOutput day "Day of the month"
+    annotation (Placement(transformation(extent={{100,30},{120,50}})));
+  Modelica.Blocks.Interfaces.IntegerOutput hour "Hour of the day"
+    annotation (Placement(transformation(extent={{100,50},{120,70}})));
+  Modelica.Blocks.Interfaces.RealOutput minute "Minute of the hour"
+    annotation (Placement(transformation(extent={{100,70},{120,90}})));
+  // fixme: why is there not a second of the minute?
```

Comment:
I think this calculation would generate too many events, hence the 'minute' is a Real instead of an integer number.